elixir/apps/web/lib/web/auth.ex

```diff
@@ -123,13 +123,9 @@ defmodule Web.Auth do
         identity_provider_identifier: identity.provider_identifier
       }
       |> Enum.reject(&is_nil(elem(&1, 1)))
-      |> URI.encode_query()
```

Comment:
This is not required but as a bonus it would be cool if we can move some of the complexity of this function to the new redirect controller.

Reply:
When you say "complexity of this function" are you referring to all of the different definitions of the `signed_in_redirect` function or to this specific instance of it?

Reply:
There is matching for context type here and validations of redirect params, if we can move some of it over I think that might be beneficial

Reply:
In order to do that I'd also need to move (at a minimum) this function over too right?

```
defp signed_in_redirect(conn, _identity, %Auth.Context{type: :client}, _encoded_fragment, _params) do
    conn
    |> Phoenix.Controller.put_flash(:error, "Please use a client application to access Firezone.")
    |> Phoenix.Controller.redirect(to: ~p"/#{conn.path_params["account_id_or_slug"]}")
    |> Plug.Conn.halt()
end
```

Otherwise, the function here would always match correct?